### PR TITLE
sat change to get value from intent

### PR DESCRIPTION
**Pull request by Nandana-NNR**
# Description
Enter a detailed description of the PR here, including the nature and purpose of the changes.

## JIRA Ticket
Include a link to the JIRA Ticket linked to this PR.
[Issue-0000](https://your-jira-domain.com/browse/Issue-0000)

## Additional Info (Test steps / Steps to reproduce etc.)
Details on how to test the feature

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -427,6 +426,10 @@ export default class App extends Base {
           if (query.params.pubSubUrl) {
             process.env.PUB_SUB_URL = query.params.pubSubUrl;
           }
+          // Set the pubSub token if present
+          if (query.params.pubSubToken) {
+            process.env.PUB_SUB_TOKEN = query.params.pubSubToken;
+          }
 
           if (query.task) {
```
Comment (neeradanelxsi) [inline src/App.js:434]: Ability to accept SAT (as FCA is open source use pubSubToken) in FCA

In intent via Discovery.navigateTo or Parameters.initialization
in urlParams


all 3 points are not addressed

Comment (neeradanelxsi) [reply src/App.js:434]: in URL https://github.com/comcast-firebolt/fca-parent/pull/414/files#diff-e41fe23dd609642313d2b1baec982c44f3e2c3cbfe5d42f4d23f7813c65118b1R152 , is this the value assigned ?

Comment (Nandana-NNR) [reply src/App.js:434]: Regarding 1st point, 
both parameters.initialisation and discovery.navigateto is handled inside this same function if scrolled up
```
 async getParameterInitializationValues() {
    await handleAsyncFunction(FireboltExampleInvoker.get().invoke(CONSTANTS.CORE.toLowerCase(), 'Parameters.initialization', [], [])).then((res) => {
      console.log('Response of Initialization :: ', res);
      if (res != undefined) {
        const action = res[0].discovery.navigateTo.action;
        if (action == 'search') {
          let query = res[0].discovery.navigateTo.data.query;
```
There is no other place in fca where discovery.navigateto is called to extract intent params. I cross-checked with Preethi, same is the way support to get pubSubUrl in intent was added
As part of this ticket, support was added to get pubSubToken via intent as support to get pubSubUrl is already implemented

Regarding point 2, will confirm after testing

Comment (neeradanelxsi) [reply src/App.js:439]: discovery.onNavigateTo is not called in the above function. Only parameters.initialization


parameters.initialization is for cold launch
discovery.onNavigateTo is for hot launch 
so if you see here, https://github.com/rdkcentral/firebolt-certification-app/blob/f8e67883d38b4d9adbde1d8fe6ce99332af9e8f6/src/LifeCycleHistory.js#L102 , we are listening to this event and extracting values from message 
https://github.com/rdkcentral/firebolt-certification-app/blob/f8e67883d38b4d9adbde1d8fe6ce99332af9e8f6/src/LifeCycleHistory.js#L113

Comment (Nandana-NNR) [reply src/App.js:439]: Have added the logic to extract and store sat here as well..but i do see this logic is executed if testoken present
 ```
if (query.params && query.params.appId && query.params.testtoken && query.params.macaddress) {
            // PUBSUB_CONNECTION environment variable has a pubsub client instance and calls the isConnected function to check the Websocket status.
            if (!process.env.PUBSUB_CONNECTION || (process.env.PUBSUB_CONNECTION && !process.env.PUBSUB_CONNECTION.isConnected())) {
              process.env.APP_TYPE = query.params.appType ? query.params.appType.toLowerCase() : CONSTANTS.FIREBOLT_CONST;
              process.env.CURRENT_APPID = query.params.appId;
              process.env.MACADDRESS = query.params.macaddress;
              process.env.TEST_TOKEN = query.params.testtoken;
              process.env.PUB_SUB_TOKEN = query.params.pubSubToken;
              const pubSubListenerCreation = new PubSubCommunication();
              const webSocketConnection = await pubSubListenerCreation.startWebSocket();
            }
          }
```
Not exactly sure if testoken is passed everytime in intent or is a required param here..so what happens if testtoken is not passed in intent..as in do we need to move the logic to get sat outside this if and place it seperately below this logic?

Comment (Nandana-NNR) [reply src/App.js:439]: Update - Removed testtoken logic as discussed

File: src/App.js [hunk 2]
```diff
@@ -434,10 +426,16 @@ export default class App extends Base {
           }
 
           // Set the pubSub URL if present
+          console.log('2507 test log - query params', query.params);
```
Comment (kevinshahfws) [inline src/App.js:429]: Please remove this before merging.

**Review by kevinshahfws — APPROVED** [review]

File: src/App.js [hunk 3]
```diff
@@ -434,10 +426,16 @@ export default class App extends Base {
...
           if (query.params.pubSubUrl) {
             process.env.PUB_SUB_URL = query.params.pubSubUrl;
+            console.log('2507 test log - query params pubsuburl', process.env.PUB_SUB_URL);
```
Comment (kschrief) [inline src/App.js:432]: Debug logs can be removed.

File: src/App.js [hunk 4]
```diff
@@ -434,10 +426,16 @@ export default class App extends Base {
...
+          }
+          // Set the pubSub token if present
+          if (query.params.pubSubToken) {
+            process.env.PUB_SUB_TOKEN = query.params.pubSubToken;
+            console.log('2507 test log - query params pubsubtoken', process.env.PUB_SUB_TOKEN);
```
Comment (kschrief) [inline src/App.js:437]: Debug logs can be removed.

File: src/App.js [hunk 5]
```diff
@@ -434,10 +426,16 @@ export default class App extends Base {
           }
 
           // Set the pubSub URL if present
+          console.log('2507 test log - query params', query.params);
           if (query.params.pubSubUrl) {
             process.env.PUB_SUB_URL = query.params.pubSubUrl;
+            console.log('2507 test log - query params pubsuburl', process.env.PUB_SUB_URL);
+          }
+          // Set the pubSub token if present
+          if (query.params.pubSubToken) {
```
Comment (kschrief) [inline src/App.js:435]: We should add an entry to search for this in the URL as well

**Review by kschrief — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 5 above)
Comment (Nandana-NNR) [reply src/App.js:435]: addressed

File: src/App.js (hunk 2 above)
Comment (neeradanelxsi) [reply src/App.js:430]: yes, will be removed once we test on sky/foxtel

File: src/App.js [hunk 6]
```diff
@@ -11,7 +11,7 @@
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
- * limitations under the License.
+ * limitations under the License
```
Comment (jbigel) [inline src/App.js:14]: Why did we remove the period?

**Review by jbigel — COMMENTED** [review]
Debug logs must be removed before merging.

**Review by jbigel — APPROVED** [review]

**Review by jbigel — APPROVED** [review]

**Pull request merged**